crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -56,66 +127,191 @@ impl TryFrom<&NovalnetRouterData<&PaymentsAuthorizeRouterData>> for NovalnetPaym
+                    house_no: item.router_data.get_optional_billing_line1(),
+                    street: item.router_data.get_optional_billing_line2(),
+                    city: item.router_data.get_optional_billing_city(),
+                    zip: item.router_data.get_optional_billing_zip(),
+                    country_code: item.router_data.get_optional_billing_country(),
+                };
+
+                let customer_ip = item
+                    .router_data
+                    .request
+                    .get_browser_info()?
+                    .get_ip_address()?;
+
+                let customer = NovalnetPaymentsRequestCustomer {
+                    first_name: item.router_data.get_billing_first_name().ok(),
+                    last_name: item.router_data.get_optional_billing_last_name(),
+                    email: item.router_data.get_optional_billing_email(),
+                    mobile: item.router_data.get_billing_phone_number().ok(),
+                    billing,
+                    customer_ip,
+                };
+
+                let lang = item
+                    .router_data
+                    .request
+                    .get_browser_info()?
+                    .get_language()
+                    .ok()
+                    .unwrap_or("EN".to_string());
+
+                let custom = NovalnetCustom { lang };
+
                 Ok(Self {
-                    amount: item.amount.clone(),
-                    card,
+                    merchant,
+                    transaction,
+                    customer,
+                    custom,
                 })
             }
             _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
 // Auth Struct
 pub struct NovalnetAuthType {
-    pub(super) api_key: Secret<String>,
+    pub(super) product_activation_key: Secret<String>,
+    pub(super) payment_access_key: Secret<String>,
+    pub(super) tariff_id: Secret<String>,
 }
 
 impl TryFrom<&ConnectorAuthType> for NovalnetAuthType {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(auth_type: &ConnectorAuthType) -> Result<Self, Self::Error> {
         match auth_type {
-            ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
+            ConnectorAuthType::SignatureKey {
+                api_key,
+                key1,
+                api_secret,
+            } => Ok(Self {
+                product_activation_key: api_key.to_owned(),
+                payment_access_key: key1.to_owned(),
+                tariff_id: api_secret.to_owned(),
             }),
             _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
         }
     }
 }
+
 // PaymentsResponse
-//TODO: Append the remaining status flags
 #[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
```

Comment:
You can also derive `Copy` on the enums with unit variants, like `NovalnetTransactionStatus`, `NovalnetAPIStatus`, `CaptureType`, etc.